### DISCUSSION TITLE
it is needed modifing a core entity in plugin

### DISCUSSION THREAD

**Discussion by lusicsmusics1** [vendurehq/vendure#1058] [General]
is it possible this clear and important feture?

**Comment by michaelbromley** [comment]
Hi,
You can use [custom fields](https://www.vendure.io/docs/developer-guide/customizing-models/) to add fields to core entities. It is also possible to use a [graphql resolver](https://www.vendure.io/docs/plugins/plugin-examples/extending-graphql-api/#add-fields-to-existing-types) to extend entities.

**Comment by lytrankieio123** [comment]
Hi @michaelbromley ,
I would like to extend from Vendure to create other features. 
In my case, I want to create blogs that are related to products. Can I do it and how to start?
Could you give me some reference docs?

Thanks in advance.

Best Regard,
Ly Tran

**Reply by michaelbromley** [reply]
Hi,
Start by getting an understanding of how plugins work in vendure: https://www.vendure.io/docs/plugins/

And to add a new feature like a blog, you can look at this example of adding a new entity: https://www.vendure.io/docs/plugins/plugin-examples/defining-db-entity/

**Reply by lytrankieio123** [reply]
Thanks for your response.

**Comment by lusicsmusics1** [comment]
@michaelbromley 
Hi again
what should i do with many2many and many2one and one2one rel 
if custom field is a entity 

https://www.vendure.io/docs/developer-guide/customizing-models/#relations
`It is possible to set up custom fields which hold references to other entities using the 'relation' type`

**Comment by lusicsmusics1** [comment]
@michaelbromley 
this error occurred 

```
The GraphQL type "mypluginEntetyX" specified by the Product.x custom field does not exist
 ```

where should i define graphql type for my entities??

**Reply by lusicsmusics1** [reply]
`graphQLType?: string: The name of the GraphQL type that corresponds to the entity.
 Can be omitted if it is the same, which is usually the case.`

**Reply by lusicsmusics1** [reply]
The GraphQL type "X" specified by the X custom field does not exist
what should i need to reform this error 
the graphql is very boring and repeated writing code without need 



![image](https://user-images.githubusercontent.com/78613940/136593959-a0ab5d15-c883-43a3-9ef5-d99de027e2c5.png)

**Reply by michaelbromley** [reply]
You can define a new type like this:

```ts
import gql from 'graphql-tag';

const schemaExtensions = gql`
  type MyNewType {
    name: String!
    ... etc
  }
`;
```

you can see an example of this here: https://github.com/vendure-ecommerce/real-world-vendure/blob/master/src/plugins/reviews/graphql/api-extensions.ts

**Reply by lusicsmusics1** [reply]
> ```ts
> import gql from 'graphql-tag';
> 
> const schemaExtensions = gql`
>   type MyNewType {
>     name: String!
>     ... etc
>   }
> `;
> ```
> 
> 
in /real-world-vendure/blob/master/src/plugins/reviews/graphql/api-extensions.ts
import is from this:
import { gql } from 'apollo-server-core';

but you say 
import gql from 'graphql-tag';
🤔🤔

**Reply by michaelbromley** [reply]
ok, I was doing it from memory. I think either import will work.

**Comment by lusicsmusics1** [comment]
there is no way we automate defining graphql type????
why we have this codes:
